File: LeetCode/cpp/303.cpp
class NumArray {
public:
    vector<int> sum;
    NumArray(vector<int> nums) {
        int size = nums.size();
        sum = vector<int>(size + 1, 0);
        for (int i = 0; i < size; ++i)
            sum[i + 1] = sum[i] + nums[i];
    }
    int sumRange(int i, int j) {
        return sum[j + 1] - sum[i];
    }
};

File: LeetCode/python3/303.py
class NumArray:
    def __init__(self, nums: List[int]):
        self.nums = nums
        self.sum = []
        cul = 0
        for i in range(len(nums)):
            cul += nums[i]
            self.sum.append(cul)

    def sumRange(self, i: int, j: int) -> int:
        return self.sum[j] - self.sum[i] + self.nums[i]
